bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -89,77 +94,177 @@
  * @author Benedikt Niehues - added events for rules
  */
 public class RuleRegistryImpl extends AbstractRegistry<Rule, String, RuleProvider>
-        implements RuleRegistry, StatusInfoCallback {
+        implements RuleRegistry, StatusInfoCallback, ManagedService {
```

Comment:
as suggested before, I would very much prefer to not implement ManagedService, but use the DS configuration mechanisms (passing them through activate() or modified() methods). Would that make sense?